src/setups/extrude_geometry.jl

```diff
@@ -0,0 +1,212 @@
+"""
+    ExtrudeGeometry(geometry; particle_spacing, direction, n_extrude=1,
+                    velocity=zeros(length(direction)),
+                    mass=nothing, density=nothing, pressure=0.0)
+Extrude either a line, a plane or a shape along a specific direction.
+
+# Arguments
+- `geometry`:           Either points defining a 3D plane (2D line)
+                        or particle coordinates defining a specific shape.
+
+# Keywords
+- `particle_spacing`:   Spacing between the particles.
+- `direction`:          Vector defining the extrusion direction.
+- `n_extrude=1`         Number of `geometry` layers in extrude direction.
+- `velocity`:           Either a function mapping each particle's coordinates to its velocity,
+                        or, for a constant fluid velocity, a vector holding this velocity.
+                        Velocity is constant zero by default.
+- `mass`:               Either `nothing` (default) to automatically compute particle mass from particle
+                        density and spacing, or a function mapping each particle's coordinates to its mass,
+                        or a scalar for a constant mass over all particles.
+- `density`:            Either a function mapping each particle's coordinates to its density,
```

Comment:
```suggestion
- `density`:            This parameter can be:

-A function that determines the density of each particle based on its coordinates,
-A constant value for uniform density across all particles.
Note: Setting this value is required if a state equation is not used and cannot be combined with the state_equation parameter.
```

Reply:
![grafik](https://github.com/trixi-framework/TrixiParticles.jl/assets/44124897/702a213b-8619-4846-b84c-97315dc7edf3)
![grafik](https://github.com/trixi-framework/TrixiParticles.jl/assets/44124897/452fe000-d6c4-4ab5-823e-a4883beb67f7)
I like it!
But let's keep the old version for now and change that in a new PR. Then we can also make the default values consistent (mentioning in text vs in the keys `mass=nothing`).

Reply:
@LasNikas Do you want to open an Issue for this?

Reply:
already done in #479